test/e2e/util/kind/setup.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	kindBinary    = flag.String("kindBinary", "kind", "path to the kind binary")
+	kubectlBinary = flag.String("kubectlBinary", "kubectl", "path to the kubectl binary")
+)
+
+// Cluster represents the running state of a KIND cluster.
+// An empty struct is enough to call Setup() on.
+type Cluster struct {
```

Comment:
I can see this becoming a ManagementCluster concept and we end up with constructors such as NewManagementClusterFromKind() or similar that allows us to keep the cluster concept and abstract out kind. I don't always want kind to be my management cluster

Reply:
I should mention this is future work, i think this is totally fine as is.